src/baseline_automatic_pipeline_system/coreference.py

```diff
@@ -54,30 +56,35 @@ def predicate_score(prop, cluster, argument_clusters, lexical_wt, argument_match
     """
 
     from eval_predicate_coref import some_word_match
-
+    implicit_mentions = 0.0
     if prop[1] != 'IMPLICIT':
         prop_terms = prop[1]
-        lexical_matches = 0
+        lexical_matches = 0.0
         for other_prop in cluster:
+            implicit_mentions = 0.0
```

Comment:
I think this is a mistake, you are zeroing it implicit_mentions counter each iteration...